### PR TITLE
Residual unroll

### PR DESCRIPTION
**Pull request by szha**
@piiswrong

### CODE REVIEW

File: python/mxnet/rnn/rnn_cell.py [hunk 1]
```diff
@@ -913,6 +913,25 @@ def __call__(self, inputs, states):
...
+    def unroll(self, length, inputs, begin_state=None, layout='NTC', merge_outputs=None):
+        self.reset()
+
+        self.base_cell._modified = False
+        outputs, states = self.base_cell.unroll(length, inputs=inputs, begin_state=begin_state,
+                                                layout=layout, merge_outputs=merge_outputs)
+        self.base_cell._modified = True
+
+        inputs, _ = _normalize_sequence(length, inputs, layout, merge_outputs)
+        print(inputs)
```
Comment (fhieber) [inline python/mxnet/rnn/rnn_cell.py]: print could be removed

Comment (szha) [inline python/mxnet/rnn/rnn_cell.py]: Thanks. Good catch.

File: python/mxnet/rnn/rnn_cell.py [hunk 2]
```diff
@@ -913,6 +913,25 @@ def __call__(self, inputs, states):
...
 
+    def unroll(self, length, inputs, begin_state=None, layout='NTC', merge_outputs=None):
+        self.reset()
+
+        self.base_cell._modified = False
+        outputs, states = self.base_cell.unroll(length, inputs=inputs, begin_state=begin_state,
+                                                layout=layout, merge_outputs=merge_outputs)
+        self.base_cell._modified = True
+
+        inputs, _ = _normalize_sequence(length, inputs, layout, merge_outputs)
```
Comment (piiswrong) [inline python/mxnet/rnn/rnn_cell.py]: no need to call _normalize_sequence

Comment (piiswrong) [inline python/mxnet/rnn/rnn_cell.py]: The handling of merge_outputs is wrong. If it's none you need to decide based on outputs.

**Pull request merged**